internals/demo.sh

```diff
@@ -4,6 +4,19 @@
 SHOW_CMD_NUMS=true
 DEMO_PROMPT="$BLUE~$COLOR_RESET$ "
 
+function check_pigz {
```

Comment:
I sadly found no (simple) POSIX compliant solution to extract zlib content (I only did a quick google search, no thorough research). 

The only thing I found was the following [stack-exchange-answer](https://unix.stackexchange.com/a/450628) which seemed a little overkill

Reply:
Thanks. I also searched around for a bit. What do you think about just using a docker setup to avoid these specific issues. There might be more to come.

Reply:
Yes that would be a good idea I think. I'm not sure if it's really required because people interested in this can easily install the dependencies but I guess more people have docker installed than `pigz` and the other dependencies.